src/main/kotlin/lotto/controller/LottoController.kt

```diff
@@ -1,33 +1,43 @@
 package lotto.controller
 
-import lotto.model.InputValidatorImpl
+import lotto.model.InputValidator
 import lotto.model.Lotto
 import lotto.model.LottoTicketGenerator
 import lotto.model.WinningStatistics
 import lotto.view.InputView
 import lotto.view.ResultView
 
-interface LottoController {
-    fun run()
-}
-
-class LottoControllerImpl(
+class LottoController(
     private val inputView: InputView,
-    private val inputValidatorImpl: InputValidatorImpl,
+    private val inputValidator: InputValidator,
     private val resultView: ResultView,
     private val lottoTicketGenerator: LottoTicketGenerator,
     private val winningStatistics: WinningStatistics,
-) : LottoController {
-    private fun processTickets(purchaseAmount: Int): Set<Lotto> {
-        val numberOfTickets = lottoTicketGenerator.generateNumberOfTickets(purchaseAmount)
-        resultView.printNumberOfTickets(numberOfTickets)
-        val tickets = lottoTicketGenerator.generateTickets(numberOfTickets)
+) {
+    private fun processTickets(
+        purchaseAmount: Int,
+        numberOfManualTickets: Int,
+    ): List<Lotto> {
+        val numberOfAutoGeneratedTickets =
+            lottoTicketGenerator.generateNumberOfTickets(purchaseAmount, numberOfManualTickets)
+        val manualTickets = inputView.readManualTickets(numberOfManualTickets)
+        val tickets =
+            lottoTicketGenerator
+                .generateTickets(
+                    manualTickets,
+                    numberOfAutoGeneratedTickets,
+                )
+        resultView
+            .printNumberOfTickets(
+                numberOfManualTickets,
+                numberOfAutoGeneratedTickets,
+            )
```

Comment:
```suggestion
        val tickets =
            lottoTicketGenerator.generateTickets(manualTickets, numberOfAutoGeneratedTickets)

        resultView.printNumberOfTickets(numberOfManualTickets, numberOfAutoGeneratedTickets)
```

Consider removing some newlines 👀

Reply:
Sorry about that. In my updated code, I have removed them.